brave/brave-core
# AIChat: Added feedback form

Resolves https://github.com/brave/brave-browser/issues/32951

This PR:
1. Adds a feedback UI form
2. Introduces a new env variable `brave_services_{dev | staging | production}_domain`

- [x] Localize strings
- [x] Send model name to API
- [x] Remove "prompt" from Rating payload
- [x] Add options to feedback select form

<img width="503" alt="image" src="https://github.com/brave/brave-core/assets/8665427/61b98c04-bb28-4273-b888-9a825694eec0">
<img width="516" alt="image" src="https://github.com/brave/brave-core/assets/8665427/35e56e31-1ef0-46f8-a125-fd28e0540457">





## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Start a conversation.
2. Hover over an assistant response; you should see a 'three-dot' button that expands.
3. Like an answer - ensure an alert is called.
4. Dislike an answer - ensure an alert is called.
5. Click on 'Add feedback' when the dislike answer alert pops up.
6. Add your feedback and click 'Send'.
